SwaggerParser/MetaDataObjectBuilder.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using ARMExplorer.Controllers;
+using Newtonsoft.Json.Linq;
+
+namespace ARMExplorer.SwaggerParser
+{
+    public class MetaDataObjectBuilder
+    {
+        private readonly ServiceDefinition _serviceDefinition;
+        private const int MaxJsonLevel = 4;
+
+        private static readonly ReadOnlyCollection<String> BlacklistedPaths = new ReadOnlyCollection<string>(new List<string>()
+        {
+            "backup",
+            "restore",
+            "backup/config",
+            "discover",
+            "metrics",
+            "repository",
+            "Clone",
+            "GetOperation",
+            "register",
+            "unregister",
+            "GetSubscription",
+            "PutSubscription",
+            "ListSubscriptionStorageAccounts",
+            "CheckDnsNameAvailability",
+            "MigrateSubscription",
+            "sharedResourceProviderBase",
+            "kuduApiName",
+            "ishostingenvironmentnameavailable",
+            "ishostnameavailable",
+            "ishostnamereservedornotallowed",
+            "isusernameavailable",
+            "premieraddonoffers",
+            "subscriptions/{subscriptionId}/providers/Microsoft.Web/publishingCredentials",
+            "webhostingplans",
+            "{subscriptionid}/resourcegroups/{resourcegroupname}/providers/microsoft.web/webhostingplan",
+            "/providers/Microsoft.Web/publishingUsers/{name}",
+            "csrs",
+            "deletedSites",
+            "/workers/{workerName}",
+            "/operations/{operationId}",
+            "/extensions/{extensionApiMethod}",
+            "/snapshots",
+            "/restorablesnapshots"
+        });
+
+        public MetaDataObjectBuilder(ServiceDefinition serviceDefinition)
+        {
+            _serviceDefinition = serviceDefinition;
+        }
+
+        private static bool StringContains(String baseString, String subString, StringComparison comparison)
```

Comment:
i initially meant to write a generic method i could use in other places but didnt have the need to use it elsewhere. so this can be inlined like you suggested.